rdt/transformers/base.py

```diff
@@ -42,8 +140,22 @@ def fit_transform(self, data):
             numpy.array:
                 Transformed data.
         """
-        self.fit(data)
-        return self.transform(data)
+        columns_data = data[self._columns]
+        return self._fit_transform(columns_data)
```

Comment:
I think that `fit_transform` could stay the same as it was (so it just calls `fit` and `transform`, and `_fit_transform` is not needed), but the `columns` argument needs to be added and passed down to `fit`